src/common/sys.cpp

```diff
@@ -103,18 +102,13 @@ void sys_pll_disable(void) {
         return;                                            //already disabled - exit
     RCC_OscInitStruct.PLL.PLLState = RCC_PLL_OFF;          //set PLL off
     RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_HSE; //set CLK source HSE
-    if (irq)
-        __disable_irq();                                                        //disable irq while switching clock
+
+    buddy::DisableInterrupts disableInterrupts;
```

Comment:
`buddy::DisableInterrupts disable_interrupts;`
and everywhere else

Reply:
fixed